retrievalmarket/impl/requestvalidation/requestvalidation.go

```diff
@@ -137,9 +139,22 @@ func (rv *ProviderRequestValidator) validatePull(receiver peer.ID, proposal *ret
 }
 
 func (rv *ProviderRequestValidator) acceptDeal(deal *retrievalmarket.ProviderDealState) (retrievalmarket.DealStatus, error) {
+	pieceInfo, isUnsealed, err := rv.env.GetPiece(deal.PayloadCID, deal.PieceCID)
+	if err != nil {
+		if err == retrievalmarket.ErrNotFound {
+			return retrievalmarket.DealStatusDealNotFound, err
+		}
+		return retrievalmarket.DealStatusErrored, err
+	}
+
+	ask, err := rv.env.GetAsk(context.TODO(), deal.PayloadCID, deal.PieceCID, pieceInfo, isUnsealed, deal.Receiver)
```

Comment:
Is it possible to use a context that is tied into the life cycle of the request? It seems like this call could hang forever.